File: CSharp/Solutions/SummaryRanges.cs
namespace Solutions;

public class SummaryRanges {
    public static IList<string> SummaryRanges1(int[] nums) {
        List<string> result = [];

        for (int i = 0; i < nums.Length; i++) {
            int end = i;
            while (end + 1 < nums.Length && nums[end + 1] - nums[end] == 1) {
                end++;
            }

            if (end > i) {
                result.Add($"{nums[i]}->{nums[end]}");
            } else {
                result.Add($"{nums[end]}");
            }

            i = end;
        }

        return result;
    }
}

File: Rust/src/summary_ranges.rs
pub fn summary_ranges(nums: Vec<i32>) -> Vec<String> {
    let mut result: Vec<String> = vec![];

    let mut i = 0;
    while i < nums.len() {
        let mut end = i;
        while end + 1 < nums.len() && nums[end + 1] - nums[end] == 1 {
            end += 1;
        }

        if end > i {
            result.push(format!("{}->{}", nums[i], nums[end]));
        } else {
            result.push(format!("{}", nums[end]));
        }

        i = end + 1;
    }

    result
}
